front-end-tools/src/Main.tsx

```diff
@@ -374,6 +393,33 @@ export default function Main(props: ConnectionProps) {
                                                 const file = hTMLInputElement.files[0];
                                                 const arrayBuffer = await file.arrayBuffer();
 
+                                                // Concordium's tooling create versioned modules e.g. `.wasm.v1` now.
+                                                // Unversioned modules cannot be created by Concordium's tooling anymore.
+                                                // We assume that the uploaded module is a versioned module
+                                                // and remove the versioned 8 bytes at the beginning.
+                                                let wasmModule;
+                                                try {
+                                                    wasmModule = await WebAssembly.compile(arrayBuffer.slice(8));
+                                                } catch (e) {
+                                                    setUploadError(
+                                                        `You might have not uploaded a versioned module. Original error: ${
+                                                            (e as Error).message
+                                                        }`
```

Comment:
We could actually handle this fairly simply.

If the byte array starts with the bytes
```
pub const MAGIC_HASH: [u8; 4] = [0x00, 0x61, 0x73, 0x6D];
```

Then it is a raw Wasm module, without the versioning. Maybe that's worth adding to avoid having this error in case they somehow upload a `.wasm` file.